### PR TITLE
introducing readiness probe

### LINKED ISSUE #3648: Add readinessProbe to Helm Chart of: keptn, jmeter-service, and helm-service
**User Story**:
As a user
When I am installing Keptn I want to have a readiness probe for each pod
So that only pods which are ready to process data receive any traffic

**Details**:
The Helm Chart of Keptn defines a liveness probe as shown here: https://github.com/keptn/keptn/blob/master/installer/manifests/keptn/charts/control-plane/templates/_helpers.tpl#L65

Next to the [liveness probe](https://github.com/keptn/keptn/blob/master/installer/manifests/keptn/charts/control-plane/templates/_helpers.tpl#L65), readiness probes needs to be enabled for each pod. See: [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-readiness-probes)
Like the implementation of the liveness probe, add the readiness probe to the `_helpers.tpl` file and then to the deployments of the keptn Helm Chart. Note, that the readiness probe has to be added to the `jmeter/helm-service`

**Actions:**

- read up on the difference of liveness vs. readiness probes
- define liveness AND readiness probes on all pods/deployments (keptn, helm-service, jmeter-service)
- Adapt initialDelaySeconds of liveness/readiness probes to something that makes sense in a typical setup (e.g., 1 minute would be unreasonably high for most Keptn service, 0 seconds too low) 
  - this can be tried out by setting it to 0 seconds initially, watching the pods (`watch -n 1 kubectl -n keptn get pods`) and see how long they take until they are ready.

:exclamation: Also think about dependencies when defining those probes! mongodb-datastore depends on mongodb, many other services depend on mongodb-datastore and/or mongodb, some services depend on NATS, etc...

**Definition of Done:**
- [ ] When installing keptn (`keptn install` or `keptn install --use-case=continuous-delivery`), each deployment has a readiness/liveness probe. 
- [ ] Checked whether initialDelaySeconds for the liveness/readiness probes is appropriate
- [ ] Reasoning for delays/probes is given in the PR

### PR DESCRIPTION
**Pull request by warber**
closes #3648 

Individual startup time: `containers_ready - initialized` :

``` 
POD: api-gateway-nginx-76674c7c7b-9ct7k
    Initialized:    2021-05-26T07:27:43.000+02:00
    ContainersReady:2021-05-26T07:27:57.000+02:00
Diff: 14
--------------------
POD: api-service-6f98597f4f-6zjc8
    Initialized:    2021-05-26T07:27:27.000+02:00
    ContainersReady:2021-05-26T07:27:35.000+02:00
Diff: 8
--------------------
POD: approval-service-5c4b7699c7-mdmqn
    Initialized:    2021-05-26T07:26:51.000+02:00
    ContainersReady:2021-05-26T07:26:59.000+02:00
Diff: 8
--------------------
POD: bridge-7c9dfbfb5f-45xrk
    Initialized:    2021-05-26T07:26:40.000+02:00
    ContainersReady:2021-05-26T07:26:44.000+02:00
Diff: 4
--------------------
POD: configuration-service-5784b5fbcc-j7xbl
    Initialized:    2021-05-26T07:26:29.000+02:00
    ContainersReady:2021-05-26T07:26:35.000+02:00
Diff: 6
--------------------
POD: helm-service-5f67cf87f8-bc9rz
    Initialized:    2021-05-26T07:26:15.000+02:00
    ContainersReady:2021-05-26T07:26:20.000+02:00
Diff: 5
--------------------
POD: jmeter-service-7bfc4888ff-55t5c
    Initialized:    2021-05-26T07:26:02.000+02:00
    ContainersReady:2021-05-26T07:26:11.000+02:00
Diff: 9
--------------------
POD: keptn-nats-cluster-0
    Initialized:    2021-05-26T07:29:22.000+02:00
    ContainersReady:2021-05-26T07:29:42.000+02:00
Diff: 20
--------------------
POD: lighthouse-service-d7975dcc5-m9rj2
    Initialized:    2021-05-26T07:25:52.000+02:00
    ContainersReady:2021-05-26T07:25:59.000+02:00
Diff: 7
--------------------
POD: mongodb-6c8785bcf4-mqgrq
    Initialized:    2021-05-26T07:28:45.000+02:00
    ContainersReady:2021-05-26T07:28:53.000+02:00
Diff: 8
--------------------
POD: mongodb-datastore-56c74c7fb6-568gf
    Initialized:    2021-05-26T07:28:33.000+02:00
    ContainersReady:2021-05-26T07:28:39.000+02:00
Diff: 6
--------------------
POD: remediation-service-84ccffbc7c-jdfkr
    Initialized:    2021-05-26T07:25:40.000+02:00
    ContainersReady:2021-05-26T07:25:48.000+02:00
Diff: 8
--------------------
POD: secret-service-ff98d78-5qc8q
    Initialized:    2021-05-26T07:24:21.000+02:00
    ContainersReady:2021-05-26T07:24:28.000+02:00
Diff: 7
--------------------
POD: shipyard-controller-94898d4f8-brjnj
    Initialized:    2021-05-26T07:25:28.000+02:00
    ContainersReady:2021-05-26T07:25:37.000+02:00
Diff: 9
--------------------
POD: statistics-service-698d7d9f7f-hv82c
    Initialized:    2021-05-26T07:24:52.000+02:00
    ContainersReady:2021-05-26T07:24:59.000+02:00
Diff: 7
--------------------
``` 

Restart of all pods: 
``` 
POD: api-gateway-nginx-76674c7c7b-km7pv
    Initialized:    2021-05-26T07:33:32.000+02:00
    ContainersReady:2021-05-26T07:33:45.000+02:00
Diff: 13
--------------------
POD: api-service-6f98597f4f-4mpcx
    Initialized:    2021-05-26T07:33:32.000+02:00
    ContainersReady:2021-05-26T07:33:47.000+02:00
Diff: 15
--------------------
POD: approval-service-5c4b7699c7-h2gwh
    Initialized:    2021-05-26T07:33:32.000+02:00
    ContainersReady:2021-05-26T07:34:20.000+02:00
Diff: 48
--------------------
POD: bridge-7c9dfbfb5f-nnq6x
    Initialized:    2021-05-26T07:33:32.000+02:00
    ContainersReady:2021-05-26T07:33:39.000+02:00
Diff: 7
--------------------
POD: configuration-service-5784b5fbcc-7pxhr
    Initialized:    2021-05-26T07:33:49.000+02:00
    ContainersReady:2021-05-26T07:33:59.000+02:00
Diff: 10
--------------------
POD: helm-service-5f67cf87f8-5nwgl
    Initialized:    2021-05-26T07:33:32.000+02:00
    ContainersReady:2021-05-26T07:34:24.000+02:00
Diff: 52
--------------------
POD: jmeter-service-7bfc4888ff-xkqn9
    Initialized:    2021-05-26T07:33:32.000+02:00
    ContainersReady:2021-05-26T07:34:27.000+02:00
Diff: 55
--------------------
POD: keptn-nats-cluster-0
    Initialized:    2021-05-26T07:33:50.000+02:00
    ContainersReady:2021-05-26T07:34:06.000+02:00
Diff: 16
--------------------
POD: lighthouse-service-d7975dcc5-n5t8p
    Initialized:    2021-05-26T07:33:33.000+02:00
    ContainersReady:2021-05-26T07:34:24.000+02:00
Diff: 51
--------------------
POD: mongodb-6c8785bcf4-8mdj5
    Initialized:    2021-05-26T07:33:33.000+02:00
    ContainersReady:2021-05-26T07:33:52.000+02:00
Diff: 19
--------------------
POD: mongodb-datastore-56c74c7fb6-n4cz7
    Initialized:    2021-05-26T07:33:33.000+02:00
    ContainersReady:2021-05-26T07:34:23.000+02:00
Diff: 50
--------------------
POD: remediation-service-84ccffbc7c-t87kx
    Initialized:    2021-05-26T07:33:33.000+02:00
    ContainersReady:2021-05-26T07:34:27.000+02:00
Diff: 54
--------------------
POD: secret-service-ff98d78-pbmxc
    Initialized:    2021-05-26T07:33:33.000+02:00
    ContainersReady:2021-05-26T07:33:46.000+02:00
Diff: 13
--------------------
POD: shipyard-controller-94898d4f8-bchzr
    Initialized:    2021-05-26T07:33:33.000+02:00
    ContainersReady:2021-05-26T07:34:24.000+02:00
Diff: 51
--------------------
POD: statistics-service-698d7d9f7f-dqtck
    Initialized:    2021-05-26T07:33:33.000+02:00
    ContainersReady:2021-05-26T07:34:23.000+02:00
Diff: 50
--------------------
```

### CODE REVIEW

**Comment by christian-kreuzberger-dtx** [comment]
* readiness default 5, interval 5
* liveness default 0 (once it is ready, we don't need to wait to verify liveness), interval 5

File: shipyard-controller/main.go [hunk 1]
```diff
@@ -124,6 +131,8 @@ func main() {
 	shipyardController.AddSequenceFinishedHook(sequenceStateMaterializedView)
 
 	engine.Static("/swagger-ui", "./swagger-ui")
+
+	go keptnapi.RunHealthEndpoint("10998")
```
Comment (christian-kreuzberger-dtx) [inline shipyard-controller/main.go:135]: I don't understand, why do we have a `healthController` up there, but also run the health endpoint on a separate port?

File: statistics-service/main.go [hunk 2]
```diff
@@ -43,5 +44,6 @@ func main() {
 
 	router.Static("/swagger-ui", "./swagger-ui")
 
+	go keptnapi.RunHealthEndpoint("10998")
```
Comment (christian-kreuzberger-dtx) [inline statistics-service/main.go:47]: Can we also use shipyard-controllers logic for creating a health endpoint with gin gonic?

File: shipyard-controller/main.go (hunk 1 above)
Comment (warber) [reply shipyard-controller/main.go:135]: forgeot to delete this line, will do

File: statistics-service/main.go (hunk 2 above)
Comment (warber) [reply statistics-service/main.go:47]: sure, good idea

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM

**Pull request merged**